packages/react-charting/src/components/AreaChart/AreaChart.base.tsx

```diff
@@ -112,6 +113,12 @@ export class AreaChartBase extends React.Component<IAreaChartProps, IAreaChartSt
       isCircleClicked: false,
       nearestCircleToHighlight: null,
       isShowCalloutPending: false,
+      emptyChart: !(
+        this.props.data &&
+        this.props.data.lineChartData &&
+        this.props.data.lineChartData.length > 0 &&
+        this.props.data.lineChartData.filter(item => item.data.length === 0).length === 0
```

Comment:
Can you help me understand this line

Reply:
This line filters out any item within the lineChartData array whose data is empty. Then it checks if the length of the filtered array is 0. If non-zero, then it indicates that there are some items within the provided lineChartdata whose 'data' is empty.

Reply:
The expression is somewhat confusing and can possibly cause bugs. Can you add comment on why length === 0 means non empty.

Reply:
added a comment "filtering all items which have no data and checking if the length of the filtered array is 0 which means chart is not empty"